### ISSUE TITLE
Clipping path support

### ISSUE THREAD

**Issue by pzich** [evil-mad/axidraw#100]
Is it possible to support clipping paths from SVGs?
As an example, this is an SVG with diagonal lines that are clipped by a circle.

<img width="372" alt="Screen Shot 2021-02-08 at 6 16 14 PM" src="https://user-images.githubusercontent.com/706138/107307076-e35b7000-6a3a-11eb-96d3-18fe5f8c6f68.png">

After pulling the output SVG out of `ad.plot_run(True)` I see the complete lines drawn instead of the clipped ones.

<img width="360" alt="Screen Shot 2021-02-08 at 6 16 32 PM" src="https://user-images.githubusercontent.com/706138/107307220-21589400-6a3b-11eb-8448-24d84d3b92e5.png">

My expectation would be that the clipping path is recognized/interpreted and the paths would look something like this.

<img width="375" alt="Screen Shot 2021-02-08 at 6 16 48 PM" src="https://user-images.githubusercontent.com/706138/107307341-611f7b80-6a3b-11eb-8d16-a71d87c9859f.png">

**Comment by oskay** [comment]
It is not possible at present without using external software to perform the clipping.

We recommend starting either with [this SVG cropper](https://msurguy.github.io/svg-cropper-tool/) for simple cases or [vpype](https://github.com/abey79/vpype) with the [occult](https://github.com/LoicGoulefert/occult) plugin.

This *is* a planned feature, with work underway for a future release.

**Comment by pzich** [comment]
Excellent! Any chance that future release will also support reversing paths ( #80 ) for greater optimization? 🙏

**Comment by oskay** [comment]
Yes, path reversing (and other major improvements to plot optimization) are underway as well.

**Comment by Kenno9000** [comment]
If you are new at this and GitHub is not yet in your lexicon, try this online (full working )demo and it is an easy DL from Git to working on your desktop.
https://mitxela.com/plotterfun/
Look for Circle of Doom or Square of Dom to graphically do what you are looking for.

**Comment by jp92004** [comment]
> It is not possible at present without using external software to perform the clipping.
> 
> We recommend starting either with [this SVG cropper](https://msurguy.github.io/svg-cropper-tool/) for simple cases or [vpype](https://github.com/abey79/vpype) with the [occult](https://github.com/LoicGoulefert/occult) plugin.
> 
> This _is_ a planned feature, with work underway for a future release.

Just curious when you anticipate the clipPath feature coming. I've been looking for workarounds (trying to 'flatten' the clipPath) but without much success. Tried vpype with occult, but no joy. I'll keep trying though! I'm attaching a file to demonstrate what I'm hoping to do. Thanks!
[spart](https://user-images.githubusercontent.com/105238772/208527163-73389e07-1d91-4177-905e-b2cc0b142aea.svg)

**Comment by oskay** [comment]
> Just curious when you anticipate the clipPath feature coming

Clipping paths in particular are planned after general path hiding. The general path hiding has been implemented upstream in dev versions for about two years now, but we've run into snag after snag in the packaging process, trying to ensure (1) that everything works (2) consistently (3) across platforms. It's been a lot of work, and I'm afraid that I cannot commit to a date.


> I'll keep trying though! I'm attaching a file to demonstrate what I'm hoping to do. Thanks!

If I understand what I'm looking at, you just want to print the area inside the square? If so, you can actually the existing clipping feature at the edge of a document to achieve this. Set the document size to the print size that you want, and the AxiDraw software will automatically clip each path at those bounds.

**Comment by jp92004** [comment]
Hey, thank you so much for the quick response Windell. You guys are kind of amazing in that way! I understand about hitting snags, you have a lot of platforms to cover. I'll hold tight for that. In the meantime I'll give that work around a whirl. Thank you!

**Comment by oskay** [comment]
Hidden-line removal is now available in AxiDraw 3.9: https://github.com/evil-mad/axidraw/releases/tag/v3.9.0

**Comment by jp92004** [comment]
> Hidden-line removal is now available in AxiDraw 3.9: https://github.com/evil-mad/axidraw/releases/tag/v3.9.0

Woohoo! Will check it out today. Thank you Windell!
